presto-docker/bin/entrypoint

```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env bash -xeu
+
+# Copy config files, if they exist, to /usr/lib/presto/etc. This is necessary
+# for Kubernetes as Secrets and ConfigMaps are mounted read only.
+if [[ -d /usr/lib/presto/conf ]]; then
+    if [[ -d /usr/lib/presto/conf/catalog ]]; then
+        find /usr/lib/presto/etc/* -type f \! -name node.id -delete
+        cp -r /usr/lib/presto/conf/* /usr/lib/presto/etc/
+    else
+        find /usr/lib/presto/etc/ -maxdepth 1 -type f \! -name node.id -delete
+        cp /usr/lib/presto/conf/* /usr/lib/presto/etc/
+    fi
+fi
+
+# Replace node.id with randomly generated one,
+# but keep the same node.id on restart.
```

Comment:
So this might need a little reworking, however one thing I'm trying to capture here is to keep the node.id consistent for a given pod if possible. I believe this may be important for Raptor and/or other catalogs that rely on locating data on local disk by node.id. @electrum might be able to speak to that. 

My thought here is that the `node.id` file gets written to persistent storage, which probably means it should move to `/data/presto/` instead of it's current location. In the case where we don't have an underlying persistent disk we effectively get the behavior that we generate a new `node.id` every time the container starts.

Reply:
I think Raptor doesn't work well with remote file system (need confirmation). I'd argue you can just ignore those catalogs that rely on persistent local disk for now. Address them later only when needed.

PS. [Local volume](https://kubernetes.io/docs/concepts/storage/volumes/#local) needs complicate settings (eg. `nodeAffinity` settings and manual cleanup).

Reply:
@11xor6 analysis sounds correct to me. If we have a persistent volume, the node ID should go there to identify the node across restarts. If there's no persistent data, then it's effectively a different node every time and thus a random node ID makes sense.

The node ID is visible in places, so it's best if it's not actually random but instead represents a unique identifier within the environment. For example, Docker container ID would be a good node ID, since it is unique and is recognizable and useful for administrators. Hostname of the container is another good choice, assuming it is unique. (It seems that by default, the hostname is the container ID, so perhaps we should use that for node ID.)

Reply:
@oneonestar Raptor implements a columnar storage engine that tracks and manages data, with a copy written to local disk on one of the workers for fast access. It is expected that local machines are unreliable, so the durable copy of data is stored in an object store like S3. This "local disk" could be a persistent flash disk that physically lives elsewhere. What matters is that the latency and bandwidth is good.